flask_inertia/inertia.py

```diff
@@ -61,13 +66,29 @@ def init_app(self, app: Flask):
         """
         self.app = app
         self._shared_data = {}
-        if not hasattr(app, "extensions"):
-            app.extensions = {}
-        app.extensions["inertia"] = self
+        if isinstance(app, Flask):
+            self._init_extension(app)
+        elif isinstance(app, Blueprint):
+            blueprint = app
+            # Register the extension once the blueprint is registered
+            blueprint.record_once(self.register_blueprint)
         app.context_processor(self.context_processor)
         app.before_request(self.process_incoming_inertia_requests)
         app.after_request(self.update_redirect)
 
+    def register_blueprint(self, state: BlueprintSetupState):
+        # TODO: support multiple instances of Inertia, each on different
+        #  blueprints? Possibly by using blueprint's name as part of the
+        #  key in app.extensions. e.g:
+        #  app.extensions[f"inertia.{state.blueprint.name}"] = self
+        self._init_extension(state.app)
+
+    def _init_extension(self, app: App):
```

Comment:
I'm not sure why you differentiate `Flask` and `App` types here

Reply:
The type of `BlueprintSetupState.app` is `App`, not `Flask`. `Flask` extends from `App`. I initially tried just `Flask` but the type checker complained.

Reply:
Ok, thanks for the clarification